src/QsCompiler/CompilationManager/AssemblyLoader.cs

```diff
@@ -172,6 +172,53 @@ public static bool LoadReferencedAssembly(Uri asm, out References.Headers header
             return compilation != null && IsValidCompilation(compilation);
         }
 
+        /// <summary>
+        /// Loads any QIR byte code included as a resource from <paramref name="assemblyFile"/>.
+        /// </summary>
+        /// <param name="assemblyFile">The reader for the byte stream of a dotnet DLL from which to load the compilation.</param>
+        /// <param name="qirByteCode">The QIR byte code included as a resource of <paramref name="assemblyFile"/>.</param>
+        /// <returns>
+        /// True if <paramref name="assemblyFile"/> includes a suitable resource, false otherwise.
+        /// </returns>
+        public static bool LoadQirByteCode(
+            PEReader assemblyFile,
+            [NotNullWhen(true)] out byte[]? qirByteCode)
```

Comment:
Using `Stream` instead of `byte[]` would be more general and flexible.

Reply:
🤔Do you know of any examples of extracting resources as streams from DLL's? I'm not really sure how.

Reply:
I've changed the type to a `Stream`, but I'm not sure if I've populated the stream correctly. Could you please double check that this is correct?

Reply:
In the latest commit you already have it already in a stream, I would just copy it to the stream passed as argument rather than setting its value:
```csharp
var resourceData = image.GetContent(absResourceOffset + sizeof(int), resourceLength).ToArray();
var tempStream = new MemoryStream(resourceData);
tempStream .CopyTo(qirBytecodeStream);
```